spec/models/course/condition/assessment_spec.rb

```diff
@@ -111,62 +111,86 @@
       end
 
       context 'when there is minimum grade percentage' do
-        subject do
-          condition = create(:course_condition_assessment, minimum_grade_percentage: 60)
-          condition.assessment = assessment
-          condition
-        end
-        # This submission is graded but its grade is below the minimum grade percentage to satisfy
-        # the condition.
-        let(:below_minimum_grade_submission) do
-          submission = create(:submission, workflow_state: :graded, assessment: assessment,
-                                           user: course_user.user)
-          answers = assessment.questions.attempt(submission)
-          answers.each do |answer|
-            answer.finalise!
-            answer.grade = 0
-            answer.save!
-          end
-          submission
-        end
-        let(:submission) do
-          create(:submission, workflow_state: :graded, assessment: assessment,
-                              user: course_user.user)
+        let(:condition) do
+          create(:course_condition_assessment, minimum_grade_percentage: 60)
         end
 
-        before do
-          submission
-          below_minimum_grade_submission
+        context 'when there is no maximum grade for the assessment' do
+          context 'when the submission is not graded' do
```

Comment:
Yes it depends on whether the submission is graded. However, I think our system currently do not allow an assessment with no question to be submitted. Is there a need to handle this corner case properly at the assessment side rather than the assessment condition?